static/js/javascript.js

```diff
@@ -1,143 +1,198 @@
+            "<td>" +
+            message +
+            "</td>" +
+            "<td>" +
+            moreinfo +
+            "</td>" +
+            "<td>" +
+            code +
+            "</td>" +
+            "<td>" +
+            codeinfo +
+            "</td>" +
+            "</tr>"
+        );
+      }
+      if (data != null) {
+        result_cb(error_list);
+      }
+    }
+    //AJAX call to pylint
+    $.post(
+      "/check_code",
+      {
+        text: code
+      },
+      function(data) {
+        current_text = data;
+        //Check Text
+        check(current_text);
+        return false;
+      },
+      "json"
+    );
+  }
 
-		}
-		//AJAX call to pylint
-		$.post('/check_code', {
-	      text :  code
-	    }, function(data) {
-	    	current_text = data;
-	    	//Check Text
-	    	check(current_text);
-	    	return false;
-	    }, 'json');
-	}
+  var editor = CodeMirror.fromTextArea(document.getElementById("txt"), {
+    mode: {
+      name: "python",
+      version: 2,
+      singleLineStringErrors: false
+    },
+    lineNumbers: true,
+    indentUnit: 4,
+    matchBrackets: true,
+    lint: true,
+    styleActiveLine: true,
+    gutters: ["CodeMirror-lint-markers"],
+    lintWith: {
+      getAnnotations: CodeMirror.remoteValidator,
+      async: true,
+      check_cb: check_syntax
+    }
+  });
 
-	var editor = CodeMirror.fromTextArea(document.getElementById("txt"), {
-        mode: {name: "python",
-               version: 2,
-               singleLineStringErrors: false},
-        lineNumbers: true,
-        indentUnit: 4,
-        matchBrackets: true,
-        lint:true,
-        styleActiveLine:true,
-        gutters: ["CodeMirror-lint-markers"],
-        lintWith: {
-	        "getAnnotations": CodeMirror.remoteValidator,
-	        "async" : true,
-	        "check_cb":check_syntax
-	    },
-    });
+  //Actually Run in Python
 
-    //Actually Run in Python
-	$( "#run" ).click(function() {
-	    $.post('/run_code', {
-	      text :  editor.getValue()
-	    }, function(data) {
-	    	print_result(data);
-	    	return false;
-	    }, 'json');
+  $("#run").click(function() {
+    var editorValue = editor.getValue();
+    $.post(
+      "/run_code",
+      {
+        text: editorValue
+      },
+      function(data) {
+        print_result(data);
+        return false;
+      },
+      "json"
+    );
 
-	    function print_result(data){
-	    	document.getElementById('output').innerHTML = '';
-	    	$("#output").append("<pre>"+data+"</pre>");
-	    }
-	});
-	var exampleCode = function (id, text) {
-        $(id).click(function (e) {
-            editor.setValue(text);
-            editor.focus(); // so that F5 works, hmm
-        });
-    };
+    function print_result(data) {
+      if (/\S/.test(editorValue)) {
+        document.getElementById("output").innerHTML = "";
+        $("#output").append("<pre>" + data + "</pre>");
+      } else document.getElementById("output").innerHTML = "Write code to see something!";
```

Comment:
So, the issue occurs during the `check` function is called not when the `run` function is called. 
 
Your logic makes sense but it shouldn't be put in the print_result but rather this check should be put in `check_syntax` specifically to check if the Editor is empty or if there are no errors in outputted by the function

Thanks for the pull request and let me know if you have any questions!